scripts/linters/pylint_extensions.py

```diff
@@ -1959,6 +2021,318 @@ def process_tokens(self, tokens):
+                    outside_args_section = True
+                    args_section_end_line_num = line_num
+                if outside_args_section and token == ':':
+                    outside_function_signature_block = True
+
+            # Checking if object type is present in function signature or not.
+            if token_type == tokenize.NAME and token == 'object':
+                if not outside_args_section:
+                    object_present_inside_arg_section = True
+                elif (
+                    outside_args_section and
+                    args_section_end_line_num == line_num
+                ):
+                    object_present_inside_return_section = True
+
+            if (
+                object_present_inside_arg_section or
+                object_present_inside_return_section
+            ):
+                object_present_in_function_signature = True
+
+            if outside_function_signature_block:
+                if object_present_in_function_signature:
+                    if (
+                        object_comment_present and
+                        object_comment_line_num + 10 >= func_def_start_line
+                    ):
+                        object_comment_present = False
+                    else:
+                        self.add_message(
+                            'object-class-used', line=func_def_start_line
+                        )
+                    object_present_in_function_signature = False
+                    object_present_inside_arg_section = False
+                    object_present_inside_return_section = False
+                if token_type == tokenize.NAME and token == 'object':
+                    # Excluding the case where 2 or more objects occurred
+                    # in the same line.
+                    if object_already_encountered_line_num == line_num:
+                        continue
+                    object_already_encountered_line_num = line_num
+                    # Excluding the case when object is called:
+                    # Eg: var = object()
+                    if 'object()' in line:
+                        continue
+                    if (
+                        object_comment_present and
+                        object_comment_line_num + 10 >= line_num
+                    ):
+                        object_comment_present = False
+                    else:
+                        self.add_message(
+                            'object-class-used', line=line_num)
+
+    def check_comment_is_present_with_cast_method(self, tokens):
+        """Checks whether the cast method in a module has been documented
+        or not. If the cast method is not documented then adds an error
+        message.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+        """
+        cast_type_regex = r'^# Here we use cast because'
+        cast_comment_present = False
+        multi_line_import = False
```

Comment:
> Isn't multi_line_import necessarily negation of single_line_import ?

No, Because multi_line_import is true only when linters are in multi line import's scope:
```
from typing import (
    < multi line import scope>
)
because here we are excluding any exceptional type.

Reply:
Also name is changed PTAL!